packages/ui/src/components/navigation/stepper/index.tsx

```diff
@@ -0,0 +1,162 @@
+import { cva } from "class-variance-authority";
+import React from "react";
+import { ReactElement, useCallback } from "react";
+import { Typography } from "../../data-display";
+
+const rootStyles = cva(["cui-flex"], {
+    variants: {
+        layout: {
+            vertical: ["cui-flex-col", "cui-gap-8"],
+            horizontal: ["cui-relative", "cui-flex-row", "cui-gap-2"],
+        },
+    },
+});
+
+const stepStyles = cva(["cui-flex", "cui-items-center", "cui-gap-4"], {
+    variants: {
+        layout: {
+            vertical: [],
+            horizontal: ["cui-flex-col", "cui-w-12"],
+        },
+        clickable: {
+            true: [
+                "hover:cui-underline",
+                "hover:cui-cursor-pointer",
+                "hover:dark:cui-decoration-white",
+            ],
+        },
+    },
+});
+
+const squareStyles = cva(["cui-relative", "cui-h-3", "cui-w-3"], {
+    variants: {
+        active: {
+            true: ["cui-bg-orange"],
+            false: ["cui-bg-black"],
+        },
+        completed: {
+            true: ["cui-border", "cui-border-orange"],
+            false: [],
+        },
+    },
+});
+
+const lineStyles = cva(["cui-absolute"], {
+    variants: {
+        layout: {
+            vertical: [
+                "cui-left-1/2",
+                "cui-top-3",
+                "cui-h-12",
+                "cui-w-[1px]",
+                "-cui-translate-x-[0.5px]",
+                "-cui-translate-y-[0.5px]",
+            ],
+            horizontal: [
+                "cui-top-1/2",
+                "cui-h-[1px]",
+                "cui-w-12",
+                "-cui-translate-y-[0.5px]",
+                "cui-translate-x-[11px]",
+            ],
+        },
+        active: {
+            true: ["cui-bg-orange"],
+            false: ["cui-bg-black"],
+        },
+    },
+});
+
+type StepperLayout = "vertical" | "horizontal";
```

Comment:
I think it would make sense for us to have an `automatic` layout type automatically switching from `vertical` to `horizontal` on < `md`